### ISSUE TITLE
Leverage dynamic caching.

### ISSUE THREAD

**Issue by rolivieri** [IBM-Swift/swift-buildpack#7]

**Comment by BDHernand** [comment]
Read up on the following link:


http://blog.altoros.com/creating-a-custom-cloud-foundry-buildpack-from-scratch-whats-under-the-hood.html
http://jamesthom.as/blog/2015/03/04/cloud-foundry-custom-buildpacks/
http://jamesthom.as/blog/2015/03/04/cloud-foundry-custom-buildpacks/

**Comment by BDHernand** [comment]
https://github.com/cloudfoundry/nodejs-buildpack/blob/master/bin/compile

is the NodeJS usage of cache

**Comment by rolivieri** [comment]
This task is being validated by https://github.com/IBM-Swift/swift-buildpack/pull/24.

**Comment by rolivieri** [comment]
After testing the changes many times on Bluemix and using bosh-lite, we have determined the following:

1) The unpack contents of the swift and clang tar files is more than 1 GB of data. The time that Cloud Foundry takes to move these files from the CACHE_DIR to main storage is a very considerable (it may take over 2 mins to move this data). A re-execution of the cf push command also took longer because this files had be made available on the CACHE_DIR folder.

2) To reduce the amount of time, instead of caching the unpack folders, we also tried caching the swift and clang tar files. This reduced significantly the amount of time that Cloud Foundry needed to copy/move the files from the CACHE_DIR to the main storage.

3) However, we noticed that there was no performance gain at all when using the CACHE_DIR. In fact, we noticed the opposite. If using the CACHE_DIR, it certainly increased the execution of the first cf push by 20, 30 seconds. Therefore, we did not find a benefit in leveraging the CACHE_DIR other than having an offline copy of the files that were downloaded during the first cf push (which, again, would take 20-30 secs longer if leveraging the CACHE_DIR folder). It is also worth mentioning that we measured the time a re-execution of the cf push command took when leveraging CACHE_DIR vs when the dependencies are downloaded again: no difference in the execution time of the cf push command.  

4) Given these findings, we don't plan to leverage the CACHE_DIR since it will impact the performance of the buildpack in a negative way.
